lib/dart.yaml

```diff
@@ -1,194 +1,209 @@
-        explicit-trailing-zero: true
-    - format-comment:
-        only-doc-comments: false
-    - function-always-returns-null
-    - map-keys-ordering
-    - match-positional-field-names-on-assignment
-    - member-ordering:
-        order:
-          - public-fields
-          - private-fields
-          - public-getters
-          - private-getters
-          - public-setters
-          - private-setters
-          - constructors
-          - close-method
-          - dispose-method
-          - public-methods
-          - private-methods
-        widgets-order:
-          - constructor
-          - named-constructor
-          - const-fields
-          - final-fields
-          - init-state-method
-          - did-change-dependencies-method
-          - did-update-widget-method
-          - dispose-method
-          - private-methods
-          - overriden-public-methods
-          - build-method
-    - move-records-to-typedefs:
-        min-fields: 3
-        min-occurences: 2
-    - move-variable-closer-to-its-usage
-    - move-variable-outside-iteration
-    - newline-before-return
-    - no-boolean-literal-compare
-    - no-empty-block
-    - no-equal-arguments:
-        ignored-parameters:
-          - height
-          - width
-    - no-equal-conditions
-    - no-equal-nested-conditions
-    - no-equal-switch-case
-    - no-equal-switch-expression-cases
-    - no-equal-then-else
-    - no-magic-number:
-        allowed: [ -1, 0, 1 ]
-    - no-object-declaration
-    - prefer-addition-subtraction-assignments
-    - prefer-any-or-every
-    - prefer-correct-callback-field-name:
-        name-pattern: ^on[A-Z]+
-    - prefer-correct-error-name:
-        allowed-names: [ error, exception, failure ]
-    - prefer-correct-identifier-length:
-        max-identifier-length: 30
-        min-identifier-length: 2
-    - prefer-correct-stream-return-type
-    - prefer-declaring-const-constructor:
-        ignore-abstract: true
-    - prefer-early-return:
-        ignore-if-case: true
-    - prefer-explicit-function-type
-    - prefer-explicit-type-arguments
-    - prefer-immediate-return
-    - prefer-match-file-name:
-        exclude:
-          - "test/**"
-          - "lib/**/*wm.dart"
-    - prefer-moving-to-variable:
-        allowed-duplicated-chains: 3
-    - prefer-named-boolean-parameters:
-        ignore-single: true
-        ignore-single-boolean: false
-    - prefer-null-aware-spread
-    - prefer-parentheses-with-if-null
-    - prefer-public-exception-classes
-    - prefer-return-await
-    - prefer-simpler-patterns-null-check
-    - prefer-switch-with-enums
-    - prefer-test-matchers
-    - prefer-typedefs-for-callbacks
-    - prefer-unique-test-names
-    - prefer-wildcard-pattern
-    # Style
-    - unnecessary-trailing-comma: false
+    rules:
+        # https://dcm.dev/docs/rules/common
+        - arguments-ordering:
+              last:
+                  - body
+                  - child
+                  - children
```

Comment:
This order of arguments is needed for the `child`, `children` and `body` arguments come last.

Note: Text styles use the body style. Should I add an excluded to it?

Reply:
> Note: Text styles use the `body` style. Should I add an excluded to it?

You mean there is a `body` text style in theme extensions?

You can look at ignoring the rule for files that describe styles, as a rule they have `theme` and `extension` in the filename, which can be used as a reference. But in this case you will need to fix this nuance of the linter either in the template project (to maintain a unified projects structure), or for the template project add an override of this rule in analysis rule file

Reply:
Yes, I meant text extensions.
At first, I added excluding for `text_scheme` file, but then i removed it in [this commit](https://github.com/surfstudio/surf-dcm-presets/pull/13/commits/4f238f028ff812b9fd3e8a463d6993250feb6fdd)
because I also thought it would be better to do it in a template project